mmedit/edit.py

```diff
@@ -0,0 +1,258 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import os
+import warnings
+from typing import Dict, List, Optional, Union
+
+import torch
+
+from mmedit.apis.inferencers import MMEditInferencer
+from mmedit.apis.inferencers.base_mmedit_inferencer import InputsType
+from mmedit.utils import register_all_modules
+
+
+class MMEdit:
+    """MMEdit API for mmediting models inference.
+
+    Args:
+        model_name (str): Name of the editing model.
+        model_version (str): Version of a specific model.
+            Default to 'a'.
+        model_config (str): Path to the config file for the editing model.
+            Default to None.
+        model_ckpt (str): Path to the checkpoint file for the editing model.
+            Default to None.
+        config_dir (str): Path to the directory containing config files.
+            Default to 'configs/'.
+        device (torch.device): Device to use for inference. Default to 'cuda'.
+    """
+    inference_supported_models = {
+        # conditional models
+        'biggan': {
+            'type': 'conditional',
+            'version': {
+                'a': {
+                    'config':
+                    'biggan/biggan_2xb25-500kiters_cifar10-32x32.py',
+                    'ckpt':
+                    'ckpt/conditional/biggan_cifar10_32x32_b25x2_500k_20210728_110906-08b61a44.pth'  # noqa: E501
```

Comment:
Will we support downloading from URL later?

Reply:
yes, and more models of each kind of inferencer should be supported